cli/azd/pkg/graphsdk/application_request_builders_test.go

```diff
@@ -0,0 +1,196 @@
+package graphsdk_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/convert"
+	"github.com/azure/azure-dev/cli/azd/pkg/graphsdk"
+	"github.com/azure/azure-dev/cli/azd/test/mocks"
+	graphsdk_mocks "github.com/azure/azure-dev/cli/azd/test/mocks/graphsdk"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetApplicationList(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		expected := []graphsdk.Application{
+			{
+				Id:          convert.RefOf("1"),
+				DisplayName: "App 1",
+			},
+			{
+				Id:          convert.RefOf("2"),
+				DisplayName: "App 2",
+			},
+		}
+
+		mockContext := mocks.NewMockContext(context.Background())
+		graphsdk_mocks.RegisterApplicationListMock(mockContext, http.StatusOK, expected)
+
+		client, err := graphsdk_mocks.CreateGraphClient(mockContext)
+		require.NoError(t, err)
+
+		apps, err := client.Applications().Get(*mockContext.Context)
+		require.NoError(t, err)
+		require.NotNil(t, apps)
+		require.Equal(t, expected, apps.Value)
+	})
+
+	t.Run("Error", func(t *testing.T) {
+		mockContext := mocks.NewMockContext(context.Background())
+		graphsdk_mocks.RegisterApplicationListMock(mockContext, http.StatusUnauthorized, nil)
+
+		client, err := graphsdk_mocks.CreateGraphClient(mockContext)
+		require.NoError(t, err)
+
+		res, err := client.Applications().Get(*mockContext.Context)
+		require.Error(t, err)
+		require.Nil(t, res)
+	})
+}
+
+func TestGetApplicationById(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		expected := graphsdk.Application{
+			Id:                  convert.RefOf("1"),
+			AppId:               convert.RefOf("app-1"),
+			DisplayName:         "App 1",
+			PasswordCredentials: []*graphsdk.ApplicationPasswordCredential{},
+		}
+
+		mockContext := mocks.NewMockContext(context.Background())
+		graphsdk_mocks.RegisterApplicationItemMock(mockContext, http.StatusOK, *expected.Id, &expected)
```

Comment:
We should be mocking the http response from the httpclient mock.  The same way as we do for the sdk client responses. That way we have unit tests for the code that parsed the http response into the expected model (graphsdk.Aplication in this case)

Reply:
Yes, these helper function do setup the mocks the same way using our httpclient mock.  They are used in multiple areas and wanted to consolidate them for easier reuse.